internal/storage/data_codec.go

```diff
@@ -1007,43 +1007,38 @@ func (deleteCodec *DeleteCodec) Deserialize(blobs []*Blob) (partitionID UniqueID
 		if err != nil {
 			return InvalidUniqueID, InvalidUniqueID, nil, err
 		}
+		defer binlogReader.Close()
```

Comment:
Defer close will cause that Reader close after all blobs deserialize finished, but old codes support close Reader if a blob deserialize finished.